### PR TITLE
Restructure repository

### PR DESCRIPTION
**Pull request by noonfom**
- **Move scripts into dedicated folder**. Move most of the scripts into a dedicated scripts folder
- **Clone mbed-os in dependencies folder**. mbed-os clone is placed in a dependencies folder under root
- **Compartmentalize testing files**. Move unit testing files into tests/UNITTESTS
- **Apply consistent naming to test scripts**. Separate scripts for building and running tests: `build.sh` and `run.sh`
- **Update docs to reflect changes**. Update testing READMEs to reflect changes

The `init.sh` script must remain in the root repository as a result of how the integrated shell args. option works in .vscode/settings.json, i.e. moving it will break what was achieved by #45

### CODE REVIEW

File: .github/workflows/run_tests.yml [hunk 1]
```diff
@@ -3,10 +3,11 @@ on:
   push:
     branches:
       - main
+      - restructure-repository
```
Comment (paul-szczepanek-arm) [inline .github/workflows/run_tests.yml:6]: ????

**Review by paul-szczepanek-arm — CHANGES_REQUESTED** [review]
see if you can still build the ble experimental services example

File: .github/workflows/run_tests.yml (hunk 1 above)
Comment (noonfom) [reply .github/workflows/run_tests.yml:6]: Thanks. It's a remnant from developing on my fork. I'll remove it.

Comment (noonfom) [reply .github/workflows/run_tests.yml]: Resolved: ca05faa81f3bf07a7e62c2a2458a5764374006c2

**Comment by noonfom** [comment]
@paul-szczepanek-arm Indeed, the new dependencies folder could break [BLE_GattServer_ExperimentalServices](https://github.com/ARMmbed/mbed-os-example-ble/tree/master/BLE_GattServer_ExperimentalServices) and other similar applications. In fact, tests/UNITTESTS/LinkLoss and tests/UNITTESTS/Template could also cause the build to fail; the .mbedignore previously in tests/UNITTESTS only excluded the mbed-os clone. I added a .mbedignore in the root of the repository to exclude dependencies and tests: a67869aac0cc3883b00a5b40596c3b441d674097. There are no other application-breaking changes in this PR.

File: tests/README.md [hunk 2]
```diff
@@ -1,141 +0,0 @@
-# Testing experimental services
```
[the remark this answers is not in the document]
Comment (noonfom) [reply tests/README.md:1]: I agree. Added in ffc85ed951a5542286c1ed4ec40a4b400f4f6b33.

**Comment by noonfom** [comment]
This [force-push](https://github.com/ARMmbed/mbed-os-experimental-ble-services/compare/ffc85ed951a5542286c1ed4ec40a4b400f4f6b33..cca797cb46023356728e8075985e12257bb7ac70) resolves the merge conflicts introduced by #43.

**Pull request merged**